Master RMarkdown Document & Render Code/make_sdc_output.R

```diff
@@ -0,0 +1,238 @@
+          "emergency admissions age",
+          "bed days age",
+          "readmissions age",
+          "AE attendances age",
+          "mh bed days age"
+        )
+      ),
+      \(data, name) {
+        data |>
+          select(
+            financial_year,
+            age_group,
+            n = any_of(c("adm", "bed_days", "read_28", "attendances"))
+          ) |>
+          mutate(
+            locality = LOCALITY,
+            name = name,
+            .before = everything()
+          ) |>
+          filter(n < 10L) # Apply SDC filter (counts less than 10)
+      }
+    ) |>
+      set_names(c(
+        "Emergency_Admissions_Age",
+        "Unscheduled_Bed_Days_Age",
+        "Readmissions_Age",
+        "AE_attendances_Age",
+        "bed_days_mh_age"
+      ))
+
+    # Append locality-specific SMR01 data to the HSCP-level lists
+    smr01_based_all[[LOCALITY]] <- smr01_based_loc |> list_rbind()
+    smr01_age_all[[LOCALITY]] <- smr01_age_loc |> list_rbind()
+
+    # Clear out Unscheduled Care data objects to free up memory
+    rm(list = setdiff(ls(), c(loop_env, "LOCALITY")))
+    gc() # Run garbage collection to further free up memory
+
+
+    # **General Health Data Processing** ----
+    # Extract and filter general health data for the current locality
+    source("General Health/3. General Health Outputs.R")
+
+    stopifnot(
+      exists("ltc_multimorbidity"),
+      exists("ltc_types"),
+      exists("top5ltc_loc")
+    )
+
+    ltc_loc <- pmap(
+      list(
+        data = list(
+          ltc_multimorbidity,
+          ltc_types,
+          top5ltc_loc
+        ),
+        name = c(
+          "ltc_multimorbidity",
+          "ltc types",
+          "top5ltc"
+        )
+      ),
+      \(data, name) {
+        data |>
+          select(
+            measure = any_of(c("total_ltc", "key", "Prevalence")),
+            any_of("age_group"),
+            n = any_of(c("people", "value"))
+          ) |>
+          mutate(
+            locality = LOCALITY,
+            name = name,
+            .before = everything()
+          ) |>
+          filter(n < 10L) # Apply SDC filter (counts less than 10)
+      }
+    ) |>
+      set_names(c("Long_Term_Conditions", "LTC_Types", "Top5_LTC"))
+
+    # Append locality-specific LTC data to the HSCP-level list
+    ltc_all[[LOCALITY]] <- ltc_loc |> list_rbind()
+
+    # Clear out General Health data objects to free up memory
+    rm(list = setdiff(ls(), loop_env))
+    gc() # Run garbage collection
+  } # End of LOCALITY loop
+
+  # **Write data to Excel file** ----
+  # Save the processed data for all localities in the HSCP to an Excel file
+  save_dataframes_to_excel(
+    dataframes = map(
+      list(smr01_based_all, smr01_age_all, ltc_all),
+      bind_rows # Combine lists of dataframes into single dataframes
+    ),
+    sheet_names = c("SMR01_based", "SMR01_Age_Groups", "LTCs_Age_Groups"),
+    file_path = path(
+      lp_path,
+      "background data 2024",
+      "SDC",
```

Comment:
I will update this to go to `background data` (overwriting previous files). I'll then add a script to copy files, once checked, to the final profiles folder.